processing/src/main/java/io/druid/query/metadata/metadata/SegmentMetadataQuery.java

```diff
@@ -39,14 +48,29 @@ public SegmentMetadataQuery(
       @JsonProperty("intervals") QuerySegmentSpec querySegmentSpec,
       @JsonProperty("toInclude") ColumnIncluderator toInclude,
       @JsonProperty("merge") Boolean merge,
-      @JsonProperty("context") Map<String, Object> context
+      @JsonProperty("context") Map<String, Object> context,
+      @JsonProperty("usingDefaultInterval") Boolean useDefaultInterval
```

Comment:
The querySegmentSpec is always going to be set after the first object construction (constructor sets it to DEFAULT_INTERVAL if it's null) so this needs to be part of the json if the query is ever going to be constructed and then re-serialized. That could potentially happen if someone creates the Query object locally and then serializes it to send to the broker.